addons/mail/static/src/core/common/thread.js

```diff
@@ -380,6 +390,12 @@ export class Thread extends Component {
         this.env.services.action.doAction(actionDescription);
     }
 
+    getMessageClassName(message) {
+        return this.messageHighlight?.highlightedMessageId === message.id
+            ? "o-highlighted bg-view shadow-lg"
+            : "";
+    }
```

Comment:
What is the reasoning to move the compute of the class from the message to the thread?
Giving "class" prop or giving "is highlighted" prop does not seem very different.

Reply:
I think message should not even be aware of weather it is highlighted or not. If we move the scrolling responsibility to the thread, why shouldn't we do it for the highlight class as well? This way the message component is as simple as it can be, without any specific logic linked to the highlight feature. (class props already exists and is way more generic)